### ISSUE TITLE
Workflow Hangs when condition is false

### ISSUE THREAD

**Issue by jasontant** [frikky/Shuffle#291]
Connecting two testing nodes and putting a condition that fails between them results in a hung execution. 

**To Reproduce**
1. Create a new workflow
2. Add a Testing node
3. Name: Truth
4. Action: "Repeat back to Me"
5. Call to "True"
6. Add a second Testing node
7. Name: "ShouldNeverExecute"
8. Action: "Repeat back to Me"
9. Call:  "Should never execute"
10. Connect the first node to the second
11. Add a condition to the branch
12. Source: $Truth
13. Destination: False

**Expected behavior**
The condition should fail and execution should stop.

**Screenshots**
![image](https://user-images.githubusercontent.com/13942607/110549731-8538b000-8100-11eb-98b2-7a94aa8736c3.png)

** Debug logs **
Run the following commands and paste them
```
2021/03/09 22:53:36 [INFO] Starting execution of 0b06cef4-1404-43d4-8d1e-d54a872387d8!
2021/03/09 22:53:36 [INFO] Running POST execution with body of length 72
2021/03/09 22:53:36 [INFO] Should start execution on node c004619e-9745-4dcc-a06b-370cc4011937
2021/03/09 22:53:36 [INFO] No execution source (trigger) specified. Setting to default
2021/03/09 22:53:36 [INFO] New startnode: c004619e-9745-4dcc-a06b-370cc4011937
2021/03/09 22:53:36 [INFO] Executing ORG: e7bf5c31-e042-4a0b-a1aa-82bc7f68cb61
2021/03/09 22:53:36 [INFO] Execution: 75145771-a89d-439b-af13-20b2817eac20 should execute onprem with execution environment "Shuffle"
2021/03/09 22:53:41 [INFO] Executionrequests (Shuffle): 1
2021/03/09 22:53:44 [INFO] Set workflowexecution based on new worker (>0.8.53) for execution 75145771-a89d-439b-af13-20b2817eac20
---
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS                     PORTS                                         NAMES
2ca46bd2d7e8        frikky/shuffle:testing_1.0.0             "/bin/sh -c 'python …"   2 minutes ago       Exited (0) 2 minutes ago                                                 Testing_1-0-0_830247a8-d3ce-4954-a156-767405523810_75145771-a89d-439b-af13-20b2817eac20
7dd3c5e3791c        frikky/shuffle:testing_1.0.0             "/bin/sh -c 'python …"   2 minutes ago       Exited (0) 2 minutes ago                                                 Testing_1-0-0_c004619e-9745-4dcc-a06b-370cc4011937_75145771-a89d-439b-af13-20b2817eac20
ca72db9504ee        ghcr.io/frikky/shuffle-worker:0.8.60     "./worker"               2 minutes ago       Up 2 minutes                                                             worker-75145771-a89d-439b-af13-20b2817eac20
f32f1f4981e6        frikky/shuffle:testing_1.0.0             "/bin/sh -c 'python …"   5 minutes ago       Exited (0) 5 minutes ago                                                 Testing_1-0-0_830247a8-d3ce-4954-a156-767405523810_672cc9df-daee-4e09-9765-aa2f8cafb13b
e4c69a0e365b        frikky/shuffle:testing_1.0.0             "/bin/sh -c 'python …"   5 minutes ago       Exited (0) 5 minutes ago                                                 Testing_1-0-0_c004619e-9745-4dcc-a06b-370cc4011937_672cc9df-daee-4e09-9765-aa2f8cafb13b
57ce07523cb8        ghcr.io/frikky/shuffle-worker:0.8.60     "./worker"               5 minutes ago       Exited (3) 5 minutes ago                                                 worker-672cc9df-daee-4e09-9765-aa2f8cafb13b
2022e4b7ffd4        ghcr.io/frikky/shuffle-frontend:0.8.60   "/entrypoint.sh ngin…"   About an hour ago   Up About an hour           0.0.0.0:3001->80/tcp, 0.0.0.0:3443->443/tcp   shuffle-frontend
5d7b62403672        ghcr.io/frikky/shuffle-backend:0.8.60    "./webapp"               About an hour ago   Up About an hour           0.0.0.0:5001->5001/tcp                        shuffle-backend
d39dcf3756a5        ghcr.io/frikky/shuffle-orborus:0.8.60    "./orborus"              About an hour ago   Up About an hour                                                         shuffle-orborus
f13eb8563ee6        frikky/shuffle:database                  "gcloud beta emulato…"   About an hour ago   Up About an hour           0.0.0.0:8000->8000/tcp                        shuffle-database
```

**Example Workflow**
[Hung Workflow.zip](https://github.com/frikky/Shuffle/files/6111879/Hung.Workflow.zip)

**Comment by frikky** [comment]
Thanks! I'll check it out and fix it tonight. 

I'm guessing it's due to a crash somewhere that isn't caught. I'm working on making debuggin way easier for later.

**Comment by frikky** [comment]
Issue: a change from FAILED to SKIPPED for conditions. The worker isn't handling it properly as a result.

**Comment by frikky** [comment]
Hey! I've done some changes that should make sure workflows ALWAYS finish now. This was a DNS, Docker, Orborus and Worker issue in a weird chain effect 

Please try to update to this version and rerun: https://github.com/frikky/Shuffle/blob/launch/docker-compose.yml

**Comment by jasontant** [comment]
Updating just the docker-compose.yml file:

```
$ docker-compose up -d
WARNING: The SHUFFLE_APP_FORCE_UPDATE variable is not set. Defaulting to a blank string.
Creating network "shuffle_shuffle" with driver "bridge"
Pulling orborus (ghcr.io/frikky/shuffle-orborus:0.8.62)...
0.8.62: Pulling from frikky/shuffle-orborus
...
Status: Downloaded newer image for ghcr.io/frikky/shuffle-orborus:0.8.62
Pulling backend (ghcr.io/frikky/shuffle-backend:0.8.62)...
...
Status: Downloaded newer image for ghcr.io/frikky/shuffle-backend:0.8.62
Pulling frontend (ghcr.io/frikky/shuffle-frontend:0.8.62)...
ERROR: name unknown
```

Switching to the HEAD of the `launch` branch:

```
$ docker-compose up -d 
Creating network "shuffle_shuffle" with driver "bridge"
Pulling frontend (ghcr.io/frikky/shuffle-frontend:0.8.62)...
ERROR: name unknown
```

**Comment by frikky** [comment]
Hehe, I simply forgot to push that one

**Comment by jasontant** [comment]
61f5159 appears to resolve this issue. One additional observation on building the workflow.  I do not see the `1 condition` indicator on the branches, even though the condition is present.

**Comment by frikky** [comment]
Try refreshing the window. We'll make the visuals way better later on for that kind of stuff

**Comment by frikky** [comment]
Hey! All of this should be fixed in the 0.8.64 and 0.8.63 versions. Could you try it out again? Same compose location: https://github.com/frikky/Shuffle/blob/launch/docker-compose.yml

**Comment by jasontant** [comment]
> Hey! All of this should be fixed in the 0.8.64 and 0.8.63 versions. Could you try it out again? Same compose location: https://github.com/frikky/Shuffle/blob/launch/docker-compose.yml

I can confirm the example workflow now completes successfully. Thanks!

**Comment by frikky** [comment]
:heart:
